src/merger.ts

```diff
@@ -210,32 +216,70 @@ export class CodeModelMerger {
             if (!isNullOrUndefined(flattenedNodes) && flattenedNodes.length > 0) {
                 for (const fnode of flattenedNodes) {
                     if (!isNullOrUndefined(fnode) && !isNullOrUndefined(fnode.language)) {
-                        for (const prop of values(param.schema.properties)) {
-                            if (
-                                !isNullOrUndefined(fnode.language?.cli?.cliKey) &&
-                                fnode.language?.cli?.cliKey === prop['language']?.cli?.cliKey
-                            ) {
-                                fnode.language.python = prop['language'].python;
-                                fnode.language.cli.pythonFlattenedFrom = param;
-                                break;
-                            } else if (!isNullOrUndefined(fnode.language?.cli?.cliFlattenTrace)) {
-                                for (const trace of values(fnode.language.cli.cliFlattenTrace)) {
-                                    if (
-                                        !isNullOrUndefined(prop['language']?.cli?.cliPath) &&
-                                        trace === prop['language'].cli.cliPath
-                                    ) {
-                                        for (const p of prop['schema']?.properties) {
-                                            if (
-                                                !isNullOrUndefined(p.language?.cli?.cliKey) &&
-                                                fnode.language?.cli?.cliKey ===
-                                                    p.language?.cli?.cliKey
-                                            ) {
-                                                fnode.language.python = p.language.python;
-                                                fnode.language.cli.pythonFlattenedFrom = prop;
-                                                break;
+                        let foundProp = false;
+                        const OutLayerProp = [];
+                        const cliFlattenTrace = param.language['cli'].cliM4Path;
+                        OutLayerProp.push([param.schema, cliFlattenTrace]);
+
+                        while (!foundProp && OutLayerProp.length > 0) {
+                            const item = OutLayerProp.shift();
```

Comment:
is the type of item an array?

Reply:
Yes OutLayerProp is an array, actually this is using a level travesal algorithm to find out all the subnodes of param schema.

Reply:
each item is a pair, the first element is the node to be dealt with, the second element is the current flatten trace so that we are finding the correct node.